source/robot_model/src/Model.cpp

```diff
@@ -378,7 +378,29 @@ bool Model::in_range(const state_representation::JointState& joint_states) {
 bool Model::in_range(const Eigen::VectorXd& vector, const Eigen::VectorXd& lower_limits,
                                                     const Eigen::VectorXd& upper_limits) {
 
-  return ((vector.array() > lower_limits.array()).all() && (vector.array() < upper_limits.array()).all());
+  return ((vector.array() >= lower_limits.array()).all() && (vector.array() <= upper_limits.array()).all());
+}
+
+state_representation::JointState Model::clamp(const state_representation::JointState& joint_states) {
```

Comment:
I would actually be in favor of naming it as `clamp_in_range` to avoid confusion with the `clamp` function of the `JointState`. This way it is also clearer that it uses the range from the model.